flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java

```diff
@@ -18,176 +18,905 @@
+                        // info
+                        .testCase(
+                                TIMESTAMP_LTZ(),
+                                LocalDateTime.parse("2021-09-24T12:34:56.123456"),
+                                "2021-09-24 12:34:56.123456")
+                        .testCase(INTERVAL(YEAR()), 84, "+7-00")
+                        .testCase(INTERVAL(MONTH()), 5, "+0-05")
+                        .testCase(INTERVAL(MONTH()), 123, "+10-03")
+                        .testCase(INTERVAL(MONTH()), 12334, "+1027-10")
+                        .testCase(INTERVAL(DAY()), 10, "+0 00:00:00.010")
+                        .testCase(INTERVAL(DAY()), 123456789L, "+1 10:17:36.789")
+                        .testCase(INTERVAL(DAY()), Duration.ofHours(36), "+1 12:00:00.000")
+                        // https://issues.apache.org/jira/browse/FLINK-21456 Not supported currently
+                        // ARRAY
+                        // MULTISET
+                        // MAP
+                        // ROW
+                        // RAW
+                        .build(),
+                CastTestSpecBuilder.test(BOOLEAN(), "To BOOLEAN")
+                        .testCase(CHAR(3), "foo", null)
+                        .testCase(CHAR(4), "true", true)
+                        .testCase(VARCHAR(5), "FalsE", false)
+                        .testCase(STRING(), "Apache Flink", null)
+                        .testCase(STRING(), "TRUE", true)
+                        .testCase(STRING(), "", null)
+                        .testCase(BOOLEAN(), true, true)
+                        .testCase(BOOLEAN(), false, false)
+                        // Not supported - no fix
+                        // BINARY
+                        // VARBINARY
+                        // BYTES
+                        .testCase(DECIMAL(4, 3), 9.87, true)
+                        .testCase(DECIMAL(2, 1), 0.0, false)
+                        .testCase(TINYINT(), -125, true)
+                        .testCase(TINYINT(), 0, false)
+                        .testCase(SMALLINT(), 32767, true)
+                        .testCase(SMALLINT(), 0, false)
+                        .testCase(INT(), -12345678, true)
+                        .testCase(INT(), 0, false)
+                        .testCase(BIGINT(), 1234567891234L, true)
+                        .testCase(BIGINT(), 0, false)
+                        .testCase(FLOAT(), -123.456, true)
+                        .testCase(FLOAT(), 0, false)
+                        .testCase(DOUBLE(), 12345.67890, true)
+                        .testCase(DOUBLE(), 0, false)
+                        // Not supported - no fix
+                        // DATE
+                        // TIME
+                        // TIMESTAMP
+                        // TIMESTAMP_WITH_TIME_ZONE
+                        // TIMESTAMP_LTZ
+                        // INTERVAL(YEAR())
```

Comment:
nit: use the `LogicalTypeRoot` for such lists, in this case `INTERVAL_YEAR_MONTH` and `INTERVAL_DAY_TIME` below

Reply:
Should I remove `BYTES()` too from the unsupported lists and maybe also use the family, like for all 3 binary types: `BINARY_STRING`?

Reply:
As discussed I will keep all types and use the LogicalTypeRoot ones as reference.